src/components/Buttons/CloseChatWindowButton/CloseChatWindowButton.test.tsx

```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+import { shallow } from 'enzyme';
+
+import CloseChatWindowButton from './CloseChatWindowButton';
+import CloseChatWindowIcon from '../../../icons/CloseChatWindowIcon';
+
+import useChatContext from '../../../hooks/useChatContext/useChatContext';
+
+jest.mock('../../../hooks/useChatContext/useChatContext');
+
+const mockUseChatContext = useChatContext as jest.Mock<any>;
+
+const mockToggleChatWindow = jest.fn();
+mockUseChatContext.mockImplementation(() => ({ setIsChatWindowOpen: mockToggleChatWindow }));
+
+describe('the CloseChatWindowButton component', () => {
+  it('should render correctly when chat window is open', () => {
+    const wrapper = shallow(<CloseChatWindowButton />);
+    expect(wrapper.containsMatchingElement(<CloseChatWindowIcon />)).toEqual(true);
+  });
+
+  it('should close the chat window when clicked on', () => {
+    const wrapper = shallow(<CloseChatWindowButton />);
+    wrapper.simulate('click');
+    expect(mockToggleChatWindow).toHaveBeenCalled();
```

Comment:
Even better would be `expect(mockToggleChatWindow).toHaveBeenCalledWith(false);`

Reply:
sounds good. because i moved all of the `CloseChatWindowButton` code into the `ChatWindowHeader` component, I had to refactor a little bit. in order to 'simulate' the clicking of the 'X' button, I went with `wrapper.find('div').at(2).simulate('click');`. wasn't sure if i should be trying to access the className from materialui? let me know what you think.